src/node.cc

```diff
@@ -3833,6 +4200,13 @@ static void ParseArgs(int* argc,
 
   // Copy remaining arguments.
   const unsigned int args_left = nargs - index;
+
+  if (is_env && args_left) {
+    fprintf(stderr, "%s: non-option '%s' is not allowed in NODE_OPTIONS\n",
```

Comment:
I can change "allowed" to "supported". I don't think you want the "non-option" removed, it says why its not supported, its not an option. Please confirm.